tests/test_taskgroups.py

```diff
@@ -1759,28 +1762,98 @@ async def typetest_optional_status(
         task_status.started(1)
 
 
-@pytest.mark.skipif(
-    sys.version_info < (3, 12),
-    reason="Eager task factories require Python 3.12",
-)
-@pytest.mark.parametrize("anyio_backend", ["asyncio"])
 async def test_eager_task_factory(request: FixtureRequest) -> None:
+    ran = False
+
     async def sync_coro() -> None:
+        nonlocal ran
+        ran = True
         # This should trigger fetching the task state
         with CancelScope():  # noqa: ASYNC100
             pass
 
-    loop = asyncio.get_running_loop()
-    old_task_factory = loop.get_task_factory()
-    loop.set_task_factory(asyncio.eager_task_factory)
-    request.addfinalizer(lambda: loop.set_task_factory(old_task_factory))
+    async with create_task_group() as tg:
+        tg.start_soon(sync_coro)
+        assert not ran
+        tg.cancel_scope.cancel()
+
+
+if sys.version_info >= (3, 12):
+
+    def task_factory_loop_factory_factory(
+        task_factory: Callable[..., Any],
+    ) -> Callable[[], asyncio.AbstractEventLoop]:
+        def factory() -> asyncio.AbstractEventLoop:
+            loop = asyncio.new_event_loop()
+            loop.set_task_factory(task_factory)
+            return loop
+
+        return factory
+
+    def create_eager_task_factory(
+        custom_task_constructor: Callable[..., Any],
+    ) -> Callable[..., Any]:
+        def factory(
+            loop: Any,
+            coro: Any,
+            *,
+            name: str | None = None,
+            context: contextvars.Context | None = None,
+        ) -> Any:
+            return custom_task_constructor(
+                coro, loop=loop, name=name, context=context, eager_start=True
+            )
+
+        return factory
```

Comment:
Just what is the point of this?

Reply:
This is to make a task factory that's eager, but not the same as the built in eager task factory, so it can test the slow path that measures task factories for eagerness